src/foam/nanos/notification/Notification.js

```diff
@@ -114,5 +120,52 @@ foam.CLASS({
       name: 'slackMessage',
       documentation: 'Message to be sent to Slack if sendSlackMessage is enabled.'
     }
+  ],
+
+  methods: [
+    {
+      name: 'checkOwnership',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      type: 'Boolean',
+      javaCode: `
+        foam.nanos.auth.User user = (foam.nanos.auth.User) x.get("user");
+        if( user == null ) return false;
+
+        Long userId = user.getId();
+        String groupId = user.getGroup();
+
+        return getUserId() != userId && getGroupId() != groupId;
```

Comment:
`grroupId` property is only used as a helper to define what users should receive notifications but it is reset to null prior being saved to journals. You should only check for userId, this line will always return false